invenio_accounts/forms.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2016 CERN.
+#
+# Invenio is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Invenio is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Invenio; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Additional non-userprofile fields used during registration.
+
+Currently supported: recaptcha
+"""
+from flask_wtf import Form, RecaptchaField
+from wtforms import FormField, StringField, SubmitField
+
+
+class RegistrationFormRecaptcha(Form):
+    """Form for editing user profile."""
+
+    recaptcha = RecaptchaField()
+
+
+def confirm_register_form_factory(Form, current_app):
+    """Return confirmation for extended registration form."""
+    class ConfirmRegisterForm(Form):
+        if current_app.config.get("RECAPTCHA_PUBLIC_KEY") and \
```

Comment:
The position of condition is not correct. You should return untouched `Form` if recaptcha is not configured.